pkg/controller/scylladbcluster/sync_configmap.go

```diff
@@ -0,0 +1,88 @@
+package scylladbcluster
```

Comment:
The two files `sync_configmap.go` and `sync_secret.go` are copies of each other with (a practical equivalent of) find-replace "configmap" into "secret".

I think that DRYing out this logic far outweighs the benefit of maintaining both implementations that should require only minimal polymorphism, if any. Is there a good reason not to do that?

Reply:
It's how our most of sync_* files look like. In the past I already tried several times and failed due to generics limitations.